src/Calculator/SettingsPage.xaml.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "SettingsPage.xaml.h"
+#include "CalcViewModel/Common/AppResourceProvider.h"
+#include "CalcViewModel/Common/LocalizationService.h"
+#include "CalcViewModel/Common/LocalizationStringUtil.h"
+#include "CalcViewModel/Common/TraceLogger.h"
+
+using namespace std;
+using namespace CalculatorApp;
+using namespace CalculatorApp::Common;
+using namespace Platform;
+using namespace Windows::ApplicationModel;
+using namespace Windows::Foundation;
+using namespace Windows::System;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Controls::Primitives;
+using namespace Windows::UI::Xaml::Data;
+using namespace winrt::Windows::Storage;
+
+#ifndef BUILD_YEAR
+#define BUILD_YEAR 2021
+#endif
+
+auto resourceLoader = AppResourceProvider::GetInstance();
+auto themevalue = Windows::Storage::ApplicationData::Current -> LocalSettings -> Values -> Lookup(L"themeSetting");
+String ^ colorSetting = safe_cast<String ^>(themevalue);
+
+SettingsPage::SettingsPage()
+{
+    auto locService = LocalizationService::GetInstance();
+
+	InitializeComponent();
+
+    Language = locService->GetLanguage();
+
+    this->SetVersionString();
+    this->SetCopyrightString();
+
+    
+    if (themevalue != nullptr)
+    {
+        if (colorSetting == L"Dark")
+        {
+            SettingsDarkTheme->IsChecked = true;
+        }
+        else if (colorSetting == L"Light")
+        {
+            SettingsLightTheme->IsChecked = true;
+        }
+        else if (colorSetting == L"System")
+        {
+            SettingsSystemTheme->IsChecked = true;
+        }
+    }
+    else
+    {
+        SettingsSystemTheme->IsChecked = true;
+    }
+
+    InitializeContributeTextBlock();
+}
+
+void SettingsPage::OnNavigatedTo(Windows::UI::Xaml::Navigation::NavigationEventArgs ^ e)
+{
+    MainPage ^ mainPage = (MainPage ^) e->Parameter;
+    if (mainPage != nullptr)
+    {
+        MainPageProperty = mainPage;
+    }
+}
+
+void SettingsPage::InitializeContributeTextBlock()
+{
+    std::wstring contributeHyperlinkText  = resourceLoader->GetResourceString(L"SettingsContribute")->Data();
+
```

Comment:
Is there a reason you are not creating the link in Xaml using the hyperlink class? https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Documents.Hyperlink?view=winrt-19041

Reply:
Because of localization. See #1449 for more.